huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template.go

```diff
@@ -0,0 +1,503 @@
+		PackType:     int32(common["pack_type"].(int)),
+	}
+
+	return &commonOpts
+}
+
+func buildAudioOpts(rawAudio []interface{}) *mpc.Audio {
+	if len(rawAudio) != 1 {
+		return nil
+	}
+
+	audio := rawAudio[0].(map[string]interface{})
+	audioOpts := mpc.Audio{
+		Codec:        int32(audio["codec"].(int)),
+		SampleRate:   int32(audio["sample_rate"].(int)),
+		Channels:     int32(audio["channels"].(int)),
+		Bitrate:      utils.Int32(int32(audio["bitrate"].(int))),
+		OutputPolicy: buildAudioOutputPolicyOpts(audio["output_policy"].(string)),
+	}
+
+	return &audioOpts
+}
+
+func buildAudioOutputPolicyOpts(outputPolicy string) *mpc.AudioOutputPolicy {
+	if outputPolicy == "" {
+		return nil
+	}
+
+	var outputPolicyOpts mpc.AudioOutputPolicy
+	switch outputPolicy {
+	case "discard":
+		outputPolicyOpts = mpc.GetAudioOutputPolicyEnum().DISCARD
+	case "transcode":
+		outputPolicyOpts = mpc.GetAudioOutputPolicyEnum().TRANSCODE
+	default:
+		log.Printf("[WARN] output_policy invalid: %s", outputPolicy)
+		return nil
+	}
+
+	return &outputPolicyOpts
+}
+
+func buildVideoOpts(rawVideo []interface{}) *mpc.Video {
+	if len(rawVideo) != 1 {
+		return nil
+	}
+
+	video := rawVideo[0].(map[string]interface{})
+	videoOpts := mpc.Video{
+		Codec:              utils.Int32(int32(video["codec"].(int))),
+		Bitrate:            utils.Int32(int32(video["bitrate"].(int))),
+		Profile:            utils.Int32(int32(video["profile"].(int))),
+		Level:              utils.Int32(int32(video["level"].(int))),
+		Preset:             utils.Int32(int32(video["preset"].(int))),
+		RefFramesCount:     utils.Int32(int32(video["ref_frames_count"].(int))),
+		MaxIframesInterval: utils.Int32(int32(video["max_iframes_interval"].(int))),
+		BframesCount:       utils.Int32(int32(video["bframes_count"].(int))),
+		FrameRate:          utils.Int32(int32(video["frame_rate"].(int))),
+		Width:              utils.Int32(int32(video["width"].(int))),
+		Height:             utils.Int32(int32(video["height"].(int))),
+		BlackCut:           utils.Int32(int32(video["black_cut"].(int))),
+		OutputPolicy:       buildVideoOutputPolicyOpts(video["output_policy"].(string)),
+	}
+
+	return &videoOpts
+}
+
+func buildVideoOutputPolicyOpts(outputPolicy string) *mpc.VideoOutputPolicy {
+	if outputPolicy == "" {
+		return nil
+	}
+
+	var outputPolicyOpts mpc.VideoOutputPolicy
+	switch outputPolicy {
+	case "discard":
+		outputPolicyOpts = mpc.GetVideoOutputPolicyEnum().DISCARD
+	case "transcode":
+		outputPolicyOpts = mpc.GetVideoOutputPolicyEnum().TRANSCODE
+	default:
+		log.Printf("[WARN] output_policy invalid: %s", outputPolicy)
+		return nil
+	}
+
+	return &outputPolicyOpts
+}
+
+func resourceMPCTranscodingTemplateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcMpcV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating MPC client : %s", err)
```

Comment:
suggest to add version information to facilitate quick positioning of API problems when errors occur.

Reply:
There is only one version of MPC api for now, so I don't think the version information is needed